SnowplowTracker/SnowplowTracker/Emitter.cs

```diff
@@ -0,0 +1,473 @@
+		private int bufferSize;
+		private long byteLimitGet;
+		private long byteLimitPost;
+		private Timer emitterTimer;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="SnowplowTracker.Emitter"/> class.
+		/// </summary>
+		/// <param name="endpoint">The collector endpoint uri</param>
+		/// <param name="protocol">What protocol to send under</param>
+		/// <param name="method">What method of sending to use</param>
+		/// <param name="sendLimit">The amount of events to pull from the database per sending attempt</param>
+		/// <param name="bufferSize">The maximum amount of events to put in a POST</param>
+		/// <param name="byteLimitGet">The byte limit for a GET request</param>
+		/// <param name="byteLimitPost">The byte limit for a POST request</param>
+		public Emitter (string endpoint, HttpProtocol protocol = HttpProtocol.HTTP, HttpMethod method = HttpMethod.POST, 
+		                int sendLimit = 500, int bufferSize = 100, long byteLimitGet = 52000, long byteLimitPost = 52000) {
+			Utils.CheckArgument (!String.IsNullOrEmpty (endpoint), "Endpoint cannot be null or empty.");
+			this.endpoint = endpoint;
+			this.collectorUri = MakeCollectorUri(endpoint, protocol, method);
+			this.httpProtocol = protocol;
+			this.httpMethod = method;
+			this.sendLimit = sendLimit;
+			this.bufferSize = bufferSize;
+			this.byteLimitGet = byteLimitGet;
+			this.byteLimitPost = byteLimitPost;
+		}
+
+		/// <summary>
+		/// Adds an event payload to the database.
+		/// </summary>
+		/// <param name="payload">Payload.</param>
+		public void Add(TrackerPayload payload) {
+			this.eventStore.AddEvent (payload);
+		}
+
+		/// <summary>
+		/// Starts the Emitter; will be run every 10ms.
+		/// </summary>
+		public void Start() {
+			if (emitterTimer == null) {
+				Log.Debug("Emitter: Starting emitter.");
+				emitterTimer = new Timer(Emit, null, TIMER_INTERVAL, Timeout.Infinite );
+			}
+		}
+		
+		/// <summary>
+		/// Stops the Emitter; will also empty the queue into the database.
+		/// </summary>
+		public void Stop() {
+			if (emitterTimer != null) {
+				Log.Debug("Emitter: Stoppping emitter.");
+				emitterTimer.Change(Timeout.Infinite, Timeout.Infinite);
+				emitterTimer.Dispose();
+				emitterTimer = null;
+			}
+		}
+	
+		/// <summary>
+		/// Performs and orchestrates all sending of events from the database
+		/// and subsequent removal of events from the database.
+		/// </summary>
+		private void Emit(object state) {
+			sending = true;
+			List<EventRow> events = eventStore.GetDescEventRange (sendLimit);
+
+			// If some events were found start sending!
+			if (events.Count > 0) {
+				Log.Debug("Emitter: Event count: " + events.Count);
+				List<RequestResult> results = SendRequests (events);
+				events = null;
+
+				int success = 0;
+				int failure = 0;
+
+				foreach (RequestResult result in results) {
+					if (result.success) {
+						foreach (int id in result.rowIds) {
+							eventStore.DeleteEvent (id);
+						}
```

Comment:
Would it be more performant to do a single query deleting all successfully sent events? (I don't know enough about SQLite to answer this question myself.)